client/src/main/java/org/eclipse/hono/client/CredentialsClient.java

```diff
@@ -38,7 +38,7 @@
      * @param authId The authentication identifier used in the credentials to retrieve.
      * @return A future indicating the result of the operation.
      *         <p>
-     *         The future will succeed if a response with status 200 has been received from the
+     *         The future will succeed if a response with status 200 or 201 has been received from the
```

Comment:
.. with status 2xx has been received ... ?

Reply:
Sounds good, but would require to change the implementation. Then we would have to deal with 204 where we can not return a payload...

Reply:
Why is that? The JavaDoc explicitly refers to the status codes that are defined by the Credentials API. These do not include 204, do they?

Reply:
Yes, that makes sense.